### PR TITLE
🌱 Bump klog & klogr

### PR DESCRIPTION
**Pull request by benmoss**
**What this PR does / why we need it**:
Bumps our klog dependency. Main feature I want to get us is https://github.com/kubernetes/klog/pull/134 but the full changelog is available here: https://github.com/kubernetes/klog/compare/v1.0.0...v2.1.0

Not bumping us to 2.2.0 since it is a breaking change and controller-runtime has yet to bump its dependency.

### CODE REVIEW

**Comment by vincepri** [comment]
/hold

We should wait for controller-runtime to update before merging this dependency upgrade.

**Comment by vincepri** [comment]
See https://github.com/kubernetes-sigs/controller-runtime/pull/1011

File: test/infrastructure/docker/go.mod [hunk 1]
```diff
@@ -9,7 +9,7 @@ require (
 	k8s.io/api v0.17.7
 	k8s.io/apimachinery v0.17.7
 	k8s.io/client-go v0.17.7
-	k8s.io/klog v1.0.0
+	k8s.io/klog/v2 v2.1.0
```
Comment (detiber) [inline test/infrastructure/docker/go.mod:12]: missing corresponding go.sum update for this file?

**Comment by benmoss** [comment]
They're moving to 2.2.0 which has the breaking change, this is just to 2.1.0. We can wait either way but I'm not sure why it matters.

**Comment by vincepri** [comment]
Going to close this for now, given that we won't be addressing this upgrade in v1alpha3, and we'll need to wait for Controller Runtime. 

Related to #2954

**Pull request closed without merging**